src/V8JsiRuntime.cpp

```diff
@@ -1574,6 +1673,106 @@ v8::Local<v8::Value> V8Runtime::valueRef(const jsi::Value &value) {
   }
 }
 
+// Adoped from Node.js code
+/*static*/ V8Runtime *V8Runtime::GetCurrent(
+    v8::Local<v8::Context> context) noexcept {
+  if (/*UNLIKELY*/ context.IsEmpty()) {
+    return nullptr;
+  }
+  if (/*UNLIKELY*/
+      context->GetNumberOfEmbedderDataFields() <=
+      ContextEmbedderIndex::ContextTag) {
+    return nullptr;
+  }
+  if (/*UNLIKELY*/ (
+      context->GetAlignedPointerFromEmbedderData(
+          ContextEmbedderIndex::ContextTag) !=
+      V8Runtime::RuntimeContextTagPtr)) {
+    return nullptr;
+  }
+  return static_cast<V8Runtime *>(context->GetAlignedPointerFromEmbedderData(
+      ContextEmbedderIndex::Runtime));
+}
+
+bool V8Runtime::HasUnhandledPromiseRejection() noexcept {
+  return !!last_unhandled_promise_;
+}
+
+std::unique_ptr<UnhandledPromiseRejection>
+V8Runtime::GetAndClearLastUnhandledPromiseRejection() noexcept {

```

Comment:
>GetAndClearLastUnhandledPromiseRejection [](start = 11, length = 40)

Vladimir, i'm curious to know the motivation behind adding these .. Are there to get some tests or any specific node modules working ?

Reply:
Without this code we cannot observe any errors which happen in the Promise continuations. As an alternative we can consider custom implementation of the V8 microtasks, I just did not try it yet. What is your concern here?

Reply:
As i mentioned, my comment was more our of curiosity than a concern, that why is this needed now for napi .. I completely welcome having this code, but was not sure how will this be used .. we didn't need it with react native as wedon't use engine native promises in RN environment.. 

---
In reply to: [611882322](https://github.com/microsoft/v8-jsi/pull/51#discussion_r611882322) [](ancestors = 611882322)

Reply:
I needed it for the unit tests to run. Without this code I could not observe errors or asserts in Promise continuations. It is not part of the official NAPI API. I had to add an extension to enable it. Though, I am not 100% satisfied with this new APIs. I want to have something more fundamental and to be able to use it for other JS engines.